TCC.Lib/Process/ProcessStartHelper.cs

```diff
@@ -0,0 +1,169 @@
+
+            process.OutputDataReceived += (sender, e) =>
+            {
+                if (e.Data == null)
+                {
+                    standardOutputResults.SetResult(standardOutput);
+                }
+                else
+                {
+                    standardOutput.Add(e.Data);
+                }
+            };
+
+            process.ErrorDataReceived += (sender, e) =>
+            {
+                if (e.Data == null)
+                {
+                    standardErrorResults.SetResult(standardError);
+                }
+                else
+                {
+                    if (outputClassifier?.IsIgnorable(e.Data) == true)
+                    {
+                        return;
+                    }
+                    if (outputClassifier?.IsInfo(e.Data) == true)
+                    {
+                        standardInfos.Add(e.Data);
+                    }
+                    else
+                    {
+                        standardError.Add(e.Data);
+                    }
+                }
+            };
+
+            var tcs = new TaskCompletionSource<CommandResult>();
+
+            async void OnProcessExit(object sender, EventArgs args)
+            {
+                var stdout = await standardOutputResults.Task;
+                var stderr = await standardErrorResults.Task;
+
+                var result = new CommandResult
+                {
+                    Command = command,
+                    ExitCode = process.ExitCode,
+                    IsSuccess = standardError.Count == 0,
+                    Output = string.Join(Environment.NewLine, stdout),
+                    Errors = string.Join(Environment.NewLine, stderr),
+                    Infos = standardInfos,
+                };
+
+                tcs.TrySetResult(result);
+                process.Exited -= OnProcessExit;
+            }
+
+            process.Exited += OnProcessExit;
+
+            using (cancellationToken.Register(() =>
+            {
+                tcs.TrySetCanceled();
+                process.Exited -= OnProcessExit;
+                try
+                {
+                    KillProcessAndChildren(process.Id);
+
+                    if (!process.HasExited)
+                    {
+                        process.Kill();
+                    }
+                }
+                catch (InvalidOperationException)
+                {
+                }
+            }))
+            {
+                try
+                {
+                    if (process.Start() == false)
+                    {
+                        tcs.TrySetException(new InvalidOperationException("Failed to start process"));
+                        process.Exited -= OnProcessExit;
+                    }
+                    else
+                    {
+                        if (process.WaitForExit((int)Math.Min(timeoutLimit.TotalMilliseconds, int.MaxValue)) == false)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1125)